### ISSUE TITLE
vertical-pod-autoscaler Godeps.json referencing wrong version (?)

### ISSUE THREAD

**Issue by justinsb** [kubernetes/autoscaler#1351]
While trying out go modules with vertical-pod-autoscaler, I encountered hundreds (?) of errors like this:

```
...
go: converting Godeps/Godeps.json: stat k8s.io/apimachinery/pkg/fields@bd0b8021277293fc71a2f800f9ec5901cc676269: unknown revision bd0b8021277293fc71a2f800f9ec5901cc676269
...
```

And indeed, it seems to be a bad revision: https://github.com/kubernetes/apimachinery/commit/bd0b8021277293fc71a2f800f9ec5901cc676269

It's not a k/k revision either, so I'm not entirely sure what's going on...

**Comment by aleksandra-malinowska** [comment]
If I understand correctly, VPA's Godeps work pretty much the same as Cluster Autoscaler. What's going on there is the following:
- it has to vendor kubernetes/kubernetes (due to dependency on scheduler's code),
- k/k is not well suited to vendoring (staging etc.),
- to avoid issues with it, fake revisions are created based on desired k/k revision: [fix_gopath.sh](https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/fix_gopath.sh), [instructions for updating Godeps](https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/FAQ.md#how-can-i-update-ca-dependencies-particularly-k8siokubernetes).

Truth is, no one has come up with an idea how to fix it yet (see #877). If go modules can help, it would be great :)

**Comment by bskiba** [comment]
I'm closing this as a sad WAI, hoping the time will come when someone solves this mess.
